### DISCUSSION TITLE
How to configure Geckos to use my TURN server

### DISCUSSION THREAD

**Discussion by Merfyn007** [geckosio/geckos.io#121] [Help / Questions]
Hello, i setting up a TURN server which work fine, but i do not know how to use it with Geckos.

In the "Phaser 3 game multiplayer example" i tried it to make the following change:

```
  init() {
    this.io = geckos({
      iceServers: process.env.NODE_ENV === 'production' ? iceServers : [
        {
          urls: "turn:54.37.74.141:3478",
          username: "my_username",
          credential: "my_password"
        }
      ]
    })
    this.io.addServer(this.game.server)
  }
```

But when i'm trying to start my game on Firefox i get the message: "WebRTC: ICE failed, add a STUN server and see about:webrtc for more details"

How do you configure your own ?

Please thank you to help me.

**Comment by yandeu** [comment]
Try `turn:USERNAME:PASSWORD@TURN_IP_OR_ADDRESS:PORT`

**Reply by Merfyn007** [reply]
Where i put this ?

**Reply by yandeu** [reply]
```
geckos({
  iceServers: [ turn:USERNAME:PASSWORD@TURN_IP_OR_ADDRESS:PORT ]
})
```

**Reply by Merfyn007** [reply]
it doesn't work, it's worst, all the web browsers say: Connection failed with status code 500.

**Reply by LeandroVCastro** [reply]
Any answer for it? I'm trying too

**Reply by yandeu** [reply]
If your server has a public ip, no TURN is needed.

**Comment by Merfyn007** [comment]
Why should i configure a STUN/TURN server when it still works if i remove the "IceServers" like that:

```
  init() {
    this.io = geckos()
    this.io.addServer(this.game.server)
  }
```

It still doesn't work on Firefox but it still work for Google chrome, Opera and Brave, which are the most used web browsers so it's ok.

I don't really understand why we should configure a STUN/TURN server. I read somewhere on a forum that you said "I never made it work without a STUN server", so how does it work when i remove the IceServers and why we should need it for Geckos ?

Thank you.
